internal/integration_test.go

```diff
@@ -0,0 +1,294 @@
+	}
+
+	if !strings.Contains(out.String(), testCase.expOutput) {
+		t.Errorf("Expected output '%v', but got '%v' instead", testCase.expOutput, out.String())
+	}
+}
+
+func TestSetImageUrl(t *testing.T) {
+	initialWallpaper, err := getPreviousWallpaper()
+	if err != nil {
+		t.Fatalf("Error getting system initial wallpaper for eventual cleanup after tests: %v", err)
+	}
+	defer setWallpaper(initialWallpaper)
+
+	cleanup := cleanupImageUrlTest(t)
+	defer cleanup()
+
+	testCase := testConfig{
+		config:            NewConfig("", true, false),
+		inputConfirmation: "y\n",
+		inputImageUrl:     fmt.Sprintf("%v\n", TEST_IMAGE_URL),
+		expOutput:         "Successfully changed background image!",
+		expError:          nil,
+	}
+
+	var out bytes.Buffer
+	inputConfirmation = strings.NewReader(testCase.inputConfirmation)
+	inputImageUrl = strings.NewReader(testCase.inputImageUrl)
+	if err := BackdropAction(&out, testCase.config, []string{}); err != nil {
+		t.Errorf("Expected NO error, but got '%v' instead", err)
+	}
+
+	currentWallpaper, err := getPreviousWallpaper()
+	if err != nil {
+		t.Fatalf("Unexpected error getting wallpaper: %v", err)
+	}
+	if !strings.Contains(currentWallpaper, TEST_IMAGE_URL_NAME) {
+		t.Errorf("Expected wallpaper '%v' to be set for image url, but got '%v' instead", TEST_IMAGE_URL_NAME, currentWallpaper)
+	}
+
+	if !strings.Contains(out.String(), testCase.expOutput) {
+		t.Errorf("Expected output '%v', but got '%v' instead", testCase.expOutput, out.String())
+	}
+}
+
+func getCurrentBackdropSlideShowSettings(t *testing.T) string {
+	t.Helper()
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		t.Fatal("Got error when getting user home dir in test helper slide show settings.")
+	}
+
+	slideShowSettings := filepath.Join(homeDir, ".local", "share", "backgrounds", "backdrop_settings", "backdrop_settings.xml")
+	fileBytes, err := os.ReadFile(slideShowSettings)
+	if err != nil {
+		t.Fatalf("Got error when opening file in test helper slide show settings: %v", err)
+	}
+
+	return string(fileBytes)
+}
+
+func getExpectedBackdropSlideShowSettings(t *testing.T) string {
+	t.Helper()
+
+	slideShowSettings := "../test/testData/backdrop_settings.xml"
+	fileBytes, err := os.ReadFile(slideShowSettings)
+	if err != nil {
+		t.Fatalf("Got error when opening file in test helper slide show settings: %v", err)
+	}
+
+	return string(fileBytes)
+}
+
+func setupTempImageFile(t *testing.T) (string, func()) {
+	t.Helper()
+
+	wallpapersPath, err := getUserWallpapersPath()
+	if err != nil {
+		t.Fatalf("Error getting userpath in test setup: %v", err)
+	}
+
+	file, err := os.CreateTemp(wallpapersPath, "backdropTestFile")
+	if err != nil {
+		t.Fatalf("Error creating temp file for test setup: %v", err)
+	}
+
+	path := strings.Split(file.Name(), "/")
+	fileName := path[len(path)-1]
+
+	return fileName, func() {
+		os.Remove(file.Name())
+	}
+}
+
+func cleanupCustomPathTest(t *testing.T) func() {
```

Comment:
Since its a helper function I don't think it requires the `Test` part of the name.

Reply:
Has been addressed